modules/i3s/src/lib/parsers/parse-zip/buffer-file-provider.ts

```diff
@@ -0,0 +1,55 @@
+import {FileProvider} from './file-provider';
+
+/**
+ * Provides file data using DataView
+ */
+export class BufferFileProvider implements FileProvider {
+  /**
+   * The DataView from which data is provided
+   */
+  file: DataView;
+
+  constructor(file: DataView) {
+    this.file = file;
+  }
+
+  /**
+   * Gets an unsigned 8-bit integer (unsigned byte) at the specified byte offset from the start of the file.
+   * @param offset The offset, in byte, from the start of the file where to read the data.
+   */
+  getUint8(offset: number): Promise<number> {
+    return Promise.resolve(this.file.getUint8(offset));
+  }
+
+  /**
+   * Gets an unsigned 16-bit integer (unsigned byte) at the specified byte offset from the start of the file.
```

Comment:
```suggestion
   * Gets an unsigned 16-bit integer at the specified byte offset from the start of the file.
```
16-bit is not byte. Well known name is "short"